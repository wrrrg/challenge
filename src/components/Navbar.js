import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faClock } from '@fortawesome/free-solid-svg-icons';
import logo from '../../static/Moove_It-Logo_W.png';

function Navbar() {
  return (
    <nav className="flex justify-between mooveItNavybg">
      <div className="link mooveItPink no-underline flex items-center pa3" to="/">
        <FontAwesomeIcon className="mooveItPinkBorder ba bw1 br-100 mooveItTeal" icon={faClock} />
        <h4 className="pl1">Minutero</h4>
      </div>

      {/* Added an anchor element to the logo so that a new window with the company
      homepage will load on user click.
      Also added rel='noopener noreferrer' due to security
      risk posed by target="_blank" without this attribute. */}

      <a href="https://www.moove-it.com" target="_blank" rel="noopener noreferrer">
        <img className="mooveItNavybg pt4 mt2 h1" src={logo} alt="Moove-it logo" />
      </a>

      <div className="flex-grow pa2 flex items-center">
        <div className="shadow-hover h2 flex items-center f6 mooveItPink hover-white bg-animate no-underline pv1 ph3 ma2 br-pill b--black" />
      </div>
    </nav>
  );
}

export default Navbar;
