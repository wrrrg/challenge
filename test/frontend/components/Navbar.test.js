/* eslint-disable react/jsx-indent */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navbar from '../../../src/components/Navbar';

describe('Navbar', () => {
  it('renders without crashing', () => {
    const isLoggedIn = true;
    const onClick = jest.fn();
    const user = {};

    mount(<Navbar />);
  });

  it('contains the anchor element with appropriate href', () => {
    const wrapper = mount(<Navbar />);
    const link = wrapper.find('a');
    expect(link.prop('href')).toEqual('https://www.moove-it.com');
  });

  it('contains the anchor element with appropriate target to open new window', () => {
    const wrapper = mount(<Navbar />);
    const link = wrapper.find('a');
    expect(link.prop('target')).toEqual('_blank');
  });


  it('contains the anchor element with appropriate target to secure the target', () => {
    const wrapper = mount(<Navbar />);
    const link = wrapper.find('a');
    expect(link.prop('rel')).toEqual('noopener noreferrer');
  });

  it('persists through the link click', () => {
    const wrapper = mount(<Navbar />);
    const link = wrapper.find('a');

    // simulate clicking on the link
    link.simulate('click');

    // check if still matches jest snapshot, as in, a new page hasn't loaded
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


// Here we want to add another test to make sure our Navbar link is
// correctly redirecting to the moove-it.com page in a new window. However,
// adding the test seems much more complicated than simply adding the html
// element. Possibly redundant. Want to move on to other tickets for now,
// but want to return if time permits.

