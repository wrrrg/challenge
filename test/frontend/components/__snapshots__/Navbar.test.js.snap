// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar persists through the link click 1`] = `
<Navbar>
  <nav
    className="flex justify-between mooveItNavybg"
  >
    <div
      className="link mooveItPink no-underline flex items-center pa3"
      to="/"
    >
      <FontAwesomeIcon
        border={false}
        className="mooveItPinkBorder ba bw1 br-100 mooveItTeal"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f017",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z",
            ],
            "iconName": "clock",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 mooveItPinkBorder ba bw1 br-100 mooveItTeal"
          data-icon="clock"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon>
      <h4
        className="pl1"
      >
        Minutero
      </h4>
    </div>
    <a
      href="https://www.moove-it.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Moove-it logo"
        className="mooveItNavybg pt4 mt2 h1"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </a>
    <div
      className="flex-grow pa2 flex items-center"
    >
      <div
        className="shadow-hover h2 flex items-center f6 mooveItPink hover-white bg-animate no-underline pv1 ph3 ma2 br-pill b--black"
      />
    </div>
  </nav>
</Navbar>
`;
